reference/promotions.v3.yml

```
@@ -196,6 +196,72 @@ paths:
         '204':
           description: The deletion was successful or the specified resource does not exist.
           content: {}
+
+  '/promotions/{promotion_id}/codegen':
+    post:
+      tags:
+        - Coupon Codes (Bulk)
+      summary: Generate Bulk Coupon Codes
+      description: |-
+        Generate a batch of coupon codes for a particular bulk coupon promotion.
+        
+        **Note:**
+        * batch_size is limited to 250 codes per request for first version. If batch_size is not an integer or larger than 250, it will return a 422 error code.
+        * The default rate limit for this endpoint is 10 concurrent requests.
+      operationId: generatePromotionCodesBatch
+      parameters:
+        - $ref: '#/components/parameters/ContentType'
+        - $ref: '#/components/parameters/Accept'
+        - $ref: '#/components/parameters/PromotionIdPath'
+      requestBody:
+        content:
+          application/json:
+            schema:
+              required:
+                - batch_size
+              type: object
+              properties:
+                batch_size:
+                  type: integer
+                  description: The number of coupon codes to generate in each batch. The maximum value is 250.
+                  example: 5
+                  minimum: 1
+                  maximum: 250
+                max_uses:
+                  type: integer
+                  description: The maximum number of times each coupon code can be used. The default value is 1.
```

🍹 probably worth mentioning about the "0" value as well
```suggestion
                  description: The maximum number of times each coupon code can be used. The default value is 1. The value 0 means unlimited usage.
```

same for `max_uses_per_customer`